extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/PentahoMetadataDomainRepository.java

```diff
@@ -656,6 +796,7 @@ private void internalReloadDomains() {
           final String domainId = (String) fileMetadata.get( PROPERTY_NAME_DOMAIN_ID );
           final String type = (String) fileMetadata.get( PROPERTY_NAME_TYPE );
           final String locale = (String) fileMetadata.get( PROPERTY_NAME_LOCALE );
+          final String datasourceType = (String) fileMetadata.get( PROPERTY_NAME_DATASOURCE_TYPE );
```

Comment:
are you referring to the line below

`if ( StringUtils.isNotEmpty( datasourceType ) ) {`
?

Someone could manually modify the JCR data. Not likely, but possible. Better safe than sorry.

Reply:
The migration code should ensure that it is not empty, even if manually reset. Clearing by hand is not a scenario I would handle, unless if for throwing or logging. The other attributes don't follow such care.

Reply:
Given removal of `canMigrate`, by https://github.com/pentaho/pentaho-platform/pull/5124/commits/31008cd137fe8ee97aa521dc078372beac5de06b, there seems to be no non-exceptional scenario where `PROPERTY_NAME_DATASOURCE_TYPE` is not in the metadata map, when `PROPERTY_NAME_DOMAIN_ID` is.

Migration occurs synchronously, within the `getFileMetadataHelper(.)` call and, when failing, it would throw an _exception_ which would be caught and logged within the `for ( final RepositoryFile child : children )` loop, before continuing to the next domain.

Reply:
This is also related to / impacts the discussion on https://github.com/pentaho/pentaho-platform/pull/5124#discussion_r955983582.